### ISSUE TITLE
Training people segmentation

### ISSUE THREAD

**Issue by Liu0329** [bernard24/RIS#4]
When I run the launcher.sh in people_learning folder, showing that "module 'IoU3Criterion' not found". 
There may be some files missing ?

**Comment by bernard24** [comment]
Sorry, that was due to a wrong version of load_everything.lua, which tried to load unnecessary modules. I have now replaced it.

**Comment by Liu0329** [comment]
@bernard24 thx, another one should be replaced seems to be nn.MatchCriterion2 in the code.
Now could you provide the _GT_list.txt  _GT_class.txt, images etc to make the code runnable ?

**Comment by bernard24** [comment]
Thanks @Liu0329. You can download datasets at e.g:
[http://mscoco.org/dataset/#download](http://mscoco.org/dataset/#download) and 
[http://host.robots.ox.ac.uk/pascal/VOC/voc2012/#devkit](http://host.robots.ox.ac.uk/pascal/VOC/voc2012/#devkit)

I have added a data_utils directory in people_learning with matlab code for formatting ground truth and creating the txt files. Adapting it to other datasets should require minor changes.

**Comment by Liu0329** [comment]
Thanks @bernard24. Now I have both datasets. But I have trouble of getting the binary mask of coco images, though I know how to use its API. Another problem is in 'format_instance_labels.m'

```
list_file = 'ImageSets/Segmentation/train.txt';
...
object_im = imread(['SegmentationObject/', tline(1:end-1), '.png']);
class_im = imread(['SegmentationClass/', tline(1:end-1), '.png']);
```

Can you give the 'list_file' and the lists of 'SegmentationObject' imgs and 'SegmentationClass' imgs ?

@bernard24  OK, now I know how to get the mask of coco. May be I can provide a list :)

**Comment by bernard24** [comment]
The previous script was targeting Pascal VOC. I have added a new script that helps formatting the COCO dataset as Pascal VOC. Hope it helps.

**Comment by Liu0329** [comment]
Thanks @bernard24  Now I am using coco images to train from scratch (with pretrained fcn8_1 and fcn8_2). Based on your paper and code, only fcn8_1 layers should be fixed, other layers should be finetuned. But when I train the network, the memory is out. Since I turn on the learn_shall_we_stop. In your script, it is turned off. I am using titan X. So how did you solve that ?

**Comment by bernard24** [comment]
That is strange. I did not have any problems even when learning all components. I did not do anything special, other than checking that the GPU was not running anything else, since this may take ~11GB.
Have you tried running RIS_infer.ipynb? If so, do you have memory problems with that?

**Comment by Liu0329** [comment]
@bernard24  RIS_infer is OK. Now I reduce the time length. That may because my desktop consumes some memory.

**Comment by brisker** [comment]
@Liu0329 
Did you  finetuned on the coco_fcn_8_1.model the author provided , or you trained from scratch?

**Comment by Liu0329** [comment]
@brisker finetuned
